app.py

```diff
@@ -190,6 +190,27 @@ def launch_from_entity(self, version=None):
 
         self._launch_app(self.context, version=version)
 
+    def _translate_version_tokens(self, raw_string, version):
+        """
+        Returns string with version tokens replaced by their values. Replaces 
+        {version} and {v0}, {v1}, etc. tokens in raw_string with their values. The {v} 
+        tokens are created by splitting the version string by non-numeric chunks which 
+        can be used to specify parts of the version string by their index. For example:
+            {version} = "9.0v4"
+            {v0} = "9"
+            {v1} = "." 
+            {v2} = "0" 
+            {v3} = "v" 
+            {v4} = "4" 
+        :param raw_string: raw string with un-translated tokens 
+        :param version: version string to use for replacement tokens
+        """
+        version_tokens = re.findall(r'\d+|\D+', version)
```

Comment:
This would still work with the example you provided and would return a set of tokens like so: `['1', '.', '0', '.', '12', 'a']`. You'd still be able to pull out any piece individually. 

But... I agree this is a little clunky and I like your suggestion, as it feels much more elegant. Will give that a whirl and re-push.